### DISCUSSION TITLE
Is it possible to pass a variable or named capture group from customManagers to customDatasources?

### DISCUSSION THREAD

**Discussion by Devinwong** [renovatebot/renovate#30810] [Request Help] [answered]
### What would you like help with?

I would like help with my configuration

### How are you running Renovate?

Mend Renovate hosted app on github.com

### Please tell us more about your question or problem

My config is as follows.
```
"customManagers": [
  {
      "customType": "regex",
      "fileMatch": [
        <some json file>
      ],
      "matchStringsStrategy": "any",
      "matchStrings": [
        "\"name=(?<packageName>[^ ]+) os=ubuntu release=(?<testname>\\d{2}\\.\\d{2})\",\\s*\"versions\":\\s*\\[\\s*\"(?<currentValue>[^\"]+)\""
      ],
      "datasourceTemplate": "custom.moby-runc",
      "versioningTemplate": "semver-coerced"
    }
```
As you can see the release value is assigned to capture group named `testname`.
The regex was validated.

And my CustomDatasources is defined as,
```
"customDatasources": {
    "moby-runc": {
      "defaultRegistryUrlTemplate": "https://packages.microsoft.com/ubuntu/{{testname}}/prod/dists/testing/main/binary-amd64/Packages",
      "format": "plain",
      "transformTemplates": [
        <some jsonata query>
      ]
    }
```

I was trying to pass the `testname` to the `defaultRegistryUrlTemplate` but it can't resolve the variable.
Is there any way to do that? It seems to me only `packageName` can be passed from customManagers to customDatasources. Is it the truth?

The Renovate app trace was complaining
```
INFO: Disallowed variable name in template
{
  "varName": "testname"
  "template": "https://packages.microsoft.com/ubuntu/{{testname}}/prod/dists/jammy/main/binary-amd64/Packages"
}
```

### Logs (if relevant)

<details><summary>Logs</summary>

```

Replace this text with your logs, between the starting and ending triple backticks

```

</details>

**Comment by viceice** [comment]
you can't pass arbitrary capture groups.

**Reply by Devinwong** [reply]
Is `packageName` the only one supported?

**Reply by secustor** [reply, marked as answer]
Outside of the registryUrl, yes. 

But as these are templates you can potentionally hack something together with string templates
 https://docs.renovatebot.com/templates/#additional-handlebars-helpers

**Reply by Devinwong** [reply]
Not sure if I follow.
Do you mean something like appending the value of `testname`, which is what I need, into `packageName`, e.g., `packageName` = `somename-testnamevalue`, and then in `customDatasources`, use string operation to split them into `somename` and `testnamevalue`?

**Reply by secustor** [reply]
Yes, that is what I mean.

**Reply by Devinwong** [reply]
Thanks for your reply. It's a good workaround when there is no perfect solution yet. 🙂

**Reply by cavokz** [reply]
> you can't pass arbitrary capture groups.

It would be very nice if it was actually possible to pass arbitrary capture groups. Is there any hope to see this implemented in some future or is it a design choice?

**Reply by alexrecuenco** [reply]
I second the request to try to see if capture groups can be added somehow, currently string templating can be tough https://github.com/renovatebot/renovate/issues/10993#issuecomment-1761656906
